File: php/src/Creational/FactoryMethod/BostonSoupFactoryMethodSubclass.php
<?

# - One of Two Subclass Factory Methods

require_once 'SoupFactoryMethod.php';

class BostonSoupFactoryMethodSubclass extends SoupFactoryMethod
{

	function makeBuffetName() { return "Boston Soup Buffet"; }
	function makeClamChowder() { return new BostonClamChowder(); }
	function makeFishChowder() { return new BostonFishChowder(); }

}

File: go/creational/factorymethod/BostonSoupFactoryMethodSubclass.go
//BostonSoupFactoryMethodSubclass - One of Two Subclass Factory Methods

package factorymethod

import (
	"../common"
)

type BostonSoupFactoryMethodSubclass struct {  //implements SoupFactoryMethod {
	BaseSoupFactoryMethod
}

func NewBostonSoupFactoryMethodSubclass () *BostonSoupFactoryMethodSubclass {
	return &BostonSoupFactoryMethodSubclass{ BaseSoupFactoryMethod{} }
}

func (this *BostonSoupFactoryMethodSubclass) MakeBuffetName() string {
	return "Boston Soup Buffet"
}

func (this *BostonSoupFactoryMethodSubclass) MakeClamChowder() common.Soup {
	return common.NewBostonClamChowder()
}

func (this *BostonSoupFactoryMethodSubclass) MakeFishChowder() common.Soup {
	return common.NewBostonFishChowder()
}

